rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -924,6 +926,93 @@ void saveFederationElection_shouldSaveInStorage() {
         assertArrayEquals(federationElectionSerializedWithVote, actualAbiCallElectionSerialized);
     }
 
+    @Test
+    void saveFundTransactionUnsignedHash_preLovell700_shouldSaveInStorage() {
+
+        // Arrange
+
+        ActivationConfig.ForBlock arrowheadActivations = ActivationConfigsForTest.arrowhead600().forBlock(0L);
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        Sha256Hash expectedHash = BitcoinTestUtils.createHash(123_456_789);
+
+        federationStorageProvider.setFundTransactionUnsignedHash(expectedHash);
+
+        // Act
+
+        federationStorageProvider.save(networkParameters, arrowheadActivations);
+
+        // Assert
+
+        byte[] actualHashBytes = storageAccessor.getFromRepository(FUND_TX_HASH_UNSIGNED.getKey(), data -> data);
+
+        assertNull(actualHashBytes);
+
+    }
+
+    @Test
+    void saveFundTransactionUnsignedHash_postLovell700_shouldSaveInStorage() {
+
+        // Arrange
+
+        ActivationConfig.ForBlock lovell700Activations = ActivationConfigsForTest.lovell700().forBlock(0L);
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        Sha256Hash expectedHash = BitcoinTestUtils.createHash(123_456_789);
+
+        federationStorageProvider.setFundTransactionUnsignedHash(expectedHash);
+
+        // Act
+
+        federationStorageProvider.save(networkParameters, lovell700Activations);
+
+        // Assert
+
+        Sha256Hash actualHash = storageAccessor.getFromRepository(FUND_TX_HASH_UNSIGNED.getKey(), Sha256Hash::wrap);
+
+        assertEquals(expectedHash, actualHash);
+
+    }
+
+    @Test
+    void saveFundTransactionUnsignedHash_postLovell700AndResettingToNull_shouldSaveNullInStorage() {
```

Comment:
this seems more like an integration test, wdyt?

Reply:
Updated.